Standardize JSON Formatting

ObliviousHarmony opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This pull request updates all of the JSON files in the monorepo to use tabs instead of spaces. This fixes some instances of mixed tabs and spaces.

### How to test the changes in this Pull Request:

N/A

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

ObliviousHarmony committed:
Fixed `package.json` Data Type

The "private" key is supposed to be a
boolean.

ObliviousHarmony committed:
Converted Spaces To Tabs In JSON

Since we almost universally use tabs instead
of spaces, this commit replaces all of the spaces
with tabs in various JSON files across the monorepo.

roykho reviewed (changes requested):
There are a lot of `No new line` linting errors that needs to be fixed. Also this PR needs a changelog even if you put no need for changelog message. At least that is the process right now.

ObliviousHarmony committed:
Added Changelogs

ObliviousHarmony committed:
Added EOF Newlines

We put an extra newline at the end of all
files to detect those that have been
malformed or truncated.

ObliviousHarmony commented:
@roykho I've gone ahead and added all of the changelog files as well as the EOF newlines. I'm not sure why the GitHub UI still shows the errors but [if you check the commit](https://github.com/woocommerce/woocommerce/commit/f41258a9f2a7cb5cb8ae333507ed81ea57f6f655) you'll see I added them.

roykho reviewed (approved):
LGTM!
